clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -91,36 +91,24 @@ impl ArithmeticSideEffects {
     ) -> bool {
         const SATURATING: &[&str] = &["core", "num", "saturating", "Saturating"];
         const WRAPPING: &[&str] = &["core", "num", "wrapping", "Wrapping"];
-        let is_non_zero = |symbol: Option<Symbol>| {
+        let is_div_or_rem = matches!(op.node, hir::BinOpKind::Div | hir::BinOpKind::Rem);
+        let is_non_zero_u = |symbol: Option<Symbol>| {
             matches!(
                 symbol,
-                Some(
-                    sym::NonZeroI128
-                        | sym::NonZeroI16
-                        | sym::NonZeroI32
-                        | sym::NonZeroI64
-                        | sym::NonZeroI8
-                        | sym::NonZeroU128
-                        | sym::NonZeroU16
-                        | sym::NonZeroU32
-                        | sym::NonZeroU64
-                        | sym::NonZeroU8
-                )
+                Some(sym::NonZeroU128 | sym::NonZeroU16 | sym::NonZeroU32 | sym::NonZeroU64 | sym::NonZeroU8)
             )
         };
-        // If the RHS is NonZero*, then division or module by zero will never occur
-        if is_non_zero(type_diagnostic_name(cx, rhs_ty)) && let hir::BinOpKind::Div | hir::BinOpKind::Rem = op.node {
-            return true;
-        }
-        // For `Saturation` or `Wrapping` (RHS), all but division and module are allowed.
-        let is_div_or_rem = matches!(op.node, hir::BinOpKind::Div | hir::BinOpKind::Rem);
-        if (match_type(cx, rhs_ty, SATURATING) || match_type(cx, rhs_ty, WRAPPING)) && !is_div_or_rem {
+        let is_sat_or_wrap = |ty: Ty<'_>| match_type(cx, ty, SATURATING) || match_type(cx, ty, WRAPPING);
+
+        // If the RHS is NonZeroU*, then division or module by zero will never occur
+        if is_non_zero_u(type_diagnostic_name(cx, rhs_ty)) && is_div_or_rem {
             return true;
         }
-        // For `Saturation` or `Wrapping` (LHS), everything is allowed
-        if match_type(cx, lhs_ty, SATURATING) || match_type(cx, lhs_ty, WRAPPING) {
-            return true;
+        // `Saturation` and `Wrapping` can overflow if the RHS is zero in a division or module
+        if is_sat_or_wrap(lhs_ty) || is_sat_or_wrap(rhs_ty) {
```

Comment:
Actually it could/should only check the LHS (like before the previous PR)?

Reply:
Done